Data Structures/linkedList.java

```diff
@@ -0,0 +1,197 @@
+                    break;
+                case 6:
+                    System.out.print("Please Enter the Node Value: ");
+                    searchNode(head, sc.nextInt());
+                    break;
+                case 7:
+                    System.out.println("Program Terminated Successfully");
+                    break;
+                default:
+                    System.out.println("Invalid Input");
+            }
+        } while (choice != 7);
+
+    }
+    //  Method to create a new linked list
+    static Node createList(int value){
+        return new Node(value);
+    }
+    //    Method to add a new node to the linked list
+    static Node addNode(Node head, int data){
+        Node temp  = new Node(data);
+        Node node = head;
+        while(node.next!=null)
+        {
+            node = node.next;
+        }
+        node.next = temp;
+        return head;
+    }
+    //  Method to delete a node from the linked list
+    static void displayList(Node head){
+        Node temp =  head;
+        System.out.println("The Linked List is : ");
+        while(temp != null){
+            System.out.print(temp.data + " ");
+            temp = temp.next;
+        }
+        System.out.println();
+    }
+    //  Method to update a node in the linked list
+    static Node deleteNode(Node head,int data){
+        Node temp = head;
+        while(temp.next != null){
+            if(temp.next.data == data){
+                temp.next = temp.next.next;
+                break;
+            }
+            temp = temp.next;
+        }
+        displayList(head);
+        return head;
+    }
+    // Method of update note in linked list
+    static void updateNode(Node head,int data){
+        Node temp = head;
+        while(temp.next != null){
+            if(temp.next.data == data){
+                System.out.println("Please Enter the new value of the node: ");
+                temp.next.data = sc.nextInt();
+                break;
+            }
+            temp = temp.next;
+        }
+        displayList(head);
+    }
+    // Merge sort on Linked List
+    static Node sortList(Node head){
+        if(head==null || head.next==null){
+            return head;
+        }
+        Node mid = getMid(head);
+        Node nextToMid = mid.next;
+        mid.next = null;
+        Node left = sortList(head);
+        Node right = sortList(nextToMid);
+        return merge(left,right);
+    }
+    //    Function to get the middle node of the linked list
+    static Node getMid(Node head){
+        if(head==null){
+            return head;
+        }
+        Node slow = head;
+        Node fast = head.next;
+        while(fast != null){
+            fast = fast.next;
+            if(fast != null){
+                slow = slow.next;
+                fast = fast.next;
+            }
+        }
+        return slow;
+    }
+    //    Function to merge two sorted linked lists
+    static Node merge(Node a,Node b){
```

Comment:
Variable name.

Reply:
Okay sir, I will change to meaningful name